presto-native-execution/CMakeLists.txt

```diff
@@ -103,81 +117,88 @@ set(VELOX_BUILD_TEST_UTILS
 
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
-set(Boost_USE_MULTITHREADED TRUE)
-find_package(
-  Boost
-  1.66.0
-  REQUIRED
-  program_options
-  context
-  filesystem
-  regex
-  thread
-  system
-  date_time
-  atomic)
-include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
-
-find_package(gflags COMPONENTS shared)
+set(BOOST_INCLUDE_LIBRARIES
+    headers
+    atomic
+    context
+    date_time
+    filesystem
+    program_options
+    regex
+    system
+    thread)
+
+set_source(Boost)
+resolve_dependency(Boost 1.66.0 COMPONENTS ${BOOST_INCLUDE_LIBRARIES})
+
+set_source(gflags)
+resolve_dependency(gflags COMPONENTS shared)
 
+# TODO
 find_library(ANTLR4_RUNTIME antlr4-runtime)
 find_path(
   ANTLR4_RUNTIME_INCLUDE_DIR
   NAMES antlr4-runtime.h
   PATH_SUFFIXES antlr4-runtime)
 include_directories(SYSTEM ${ANTLR4_RUNTIME_INCLUDE_DIR})
 
-find_library(GLOG glog)
+if(${gflags_SOURCE} STREQUAL "BUNDLED")
+  # we force glog from source to avoid issues with a system version built
+  # against another gflags version (which is likely)
+  set(glog_SOURCE BUNDLED)
+else()
+  set(glog_SOURCE SYSTEM)
+endif()
+resolve_dependency(glog)
 
-find_library(FMT fmt)
+set_source(fmt)
+resolve_dependency(fmt)
 
+# TODO
 find_library(EVENT event)
 
-find_library(DOUBLE_CONVERSION double-conversion)
-
-find_library(LZ4 lz4)
-find_library(LZO lzo2)
-find_library(ZSTD zstd)
-find_package(ZLIB)
-find_library(SNAPPY snappy)
-
-find_package(folly CONFIG REQUIRED)
-set(FOLLY_WITH_DEPENDENCIES
-    ${FOLLY_LIBRARIES}
-    ${DOUBLE_CONVERSION}
-    Boost::context
-    dl
-    ${EVENT}
-    ${SNAPPY}
-    ${LZ4}
-    ${ZSTD}
-    ${ZLIB_LIBRARIES})
-
-find_package(BZip2 MODULE)
-if(BZIP2_FOUND)
-  list(APPEND FOLLY_WITH_DEPENDENCIES ${BZIP2_LIBRARIES})
-endif()
-include_directories(SYSTEM ${FOLLY_INCLUDE_DIRS})
```

Comment:
Same as in velox, see https://github.com/facebookincubator/velox/pull/4735 (also removes BZIP2 check there)